### PR TITLE
Bug 1998563: Fix machine page columns visibility

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Review by spadgett — COMMENTED** [review]
LGTM, but this can also happen on the pods tab of the nodes page. We should look at that as well.

Screen Shot 2021-08-30 at 8 11 30 AM

File: frontend/public/components/factory/Table/VirtualizedTable.tsx [hunk 1]
```diff
@@ -52,6 +52,20 @@ const isColumnVisible = <D extends any>(
   return true;
 };
 
+export type TableDataProps = {
+  id: string;
+  activeColumnIDs: Set<string>;
+  className?: string;
+};
+
+export const TableData: React.FC<TableDataProps> = ({ className, id, activeColumnIDs, children }) =>
```
Comment (rawagner) [inline frontend/public/components/factory/Table/VirtualizedTable.tsx:61]: this component replaces https://github.com/openshift/console/blob/master/frontend/public/components/factory/table.tsx#L217

the old one has some additional logic (`isColumnVisible`) which is not needed anymore as new Rows get a list of active columns so we just need to check if the `id` is there.

This new component also has less props & will be exposed as part of the dynamic plugin API

**Comment by rawagner** [comment]
@spadgett thank you, i was able to reproduce & fixed the issue

**Review by spadgett — APPROVED** [review]
👍 

/lgtm

**Pull request merged**